packages/phone-number-privacy/signer/src/server.ts

```diff
@@ -56,33 +57,32 @@ export function startSigner(
     res.send(PromClient.register.metrics())
   })
 
-  // TODO(2.0.0, metering) Clean this up / maybe roll into to Controller class
-  const addMeteredSignerEndpoint = (
+  const addEndpointWithTimeout = (
     endpoint: SignerEndpoint,
     handler: (req: Request, res: Response) => Promise<void>
   ) =>
     app.post(endpoint, async (req, res) => {
-      await callAndMeterLatency(endpoint, handler, req, res)
-    })
-
-  async function callAndMeterLatency(
-    endpoint: SignerEndpoint,
-    handler: (req: Request, res: Response) => Promise<void>,
-    req: Request,
-    res: Response
-  ) {
-    const childLogger: Logger = res.locals.logger
-    const end = Histograms.responseLatency.labels(endpoint).startTimer()
-    const timeoutRes = Symbol()
-    await timeout(handler, [req, res], config.timeout, timeoutRes)
-      .catch((error: any) => {
-        if (error === timeoutRes) {
+      const childLogger: Logger = res.locals.logger
+      const timeoutRes = Symbol()
+      try {
+        // TODO(2.0.0, timeout) https://github.com/celo-org/celo-monorepo/issues/9845
+        await timeout(handler, [req, res], config.timeout, timeoutRes)
+      } catch (err: any) {
+        // Handle any errors that otherwise managed to escape the proper handlers
+        let errorMsg: string = ErrorMessage.UNKNOWN_ERROR
+        let errToLog = err
+        if (err === timeoutRes) {
           Counters.timeouts.inc()
-          childLogger.warn(`Timed out after ${config.timeout}ms`)
+          errorMsg = ErrorMessage.TIMEOUT_FROM_SIGNER
+          errToLog = new Error(errorMsg)
         }
-      })
-      .finally(end)
-  }
+        childLogger.error({ errToLog })
+        res.status(500).json({
```

Comment:
this would also reduce the number of times we need to call .bind which is kinda nice

Reply:
From what I could find about Express error handling, it may be a good idea to try/catch & send a response in the outer-most endpoint code anyways in case something goes reaaally wrong (so we at least return the error). I would suggest we take another look at this when we 1) consider moving the timeout logic code -> controller/handler anyways (which should result in timeout errors being handled by the controller directly, leaving this outer try/catch for super unexpected other things), and 2) audit our endpoint error handling code (i.e. double check that we are returning responses in all cases, etc.). I created a ticket for (1) and a draft ticket for (2), but what do you think of that? Otherwise happy to do that as part of this ticket but I think we should do 1 & 2 in any case!!

Reply:
resolving in favor of the audit ticket